### ISSUE TITLE
APIService not available (k8s 1.22)

### ISSUE THREAD

**Issue by jayunit100** [tigera/operator#1380]
```
APIService
The apiregistration.k8s.io/v1beta1 API version of APIService will no longer be served in v1.22.

Migrate manifests and API clients to use the apiregistration.k8s.io/v1 API version, available since v1.10.
All existing persisted objects are accessible via the new API
No notable changes
```

seems like tigera operator isnt happy  on 1.22... 
is that because of a feature gate i need to enable

```vagrant@controlplane:~$ kubectl logs  tigera-operator-86c4fc874f-s9vhd -n tigera-operator
2021/07/08 14:36:49 [INFO] Version: v1.17.4
2021/07/08 14:36:49 [INFO] Go Version: go1.15.2
2021/07/08 14:36:49 [INFO] Go OS/Arch: linux/amd64
{"level":"info","ts":1625755009.7462676,"logger":"setup","msg":"Checking type of cluster","provider":""}
{"level":"info","ts":1625755009.7475083,"logger":"setup","msg":"Checking if TSEE controllers are required","required":false}
{"level":"info","ts":1625755009.856925,"logger":"setup","msg":"starting manager"}
{"level":"info","ts":1625755009.8569715,"logger":"typha_autoscaler","msg":"Starting typha autoscaler","syncPeriod":10}
I0708 14:36:49.858270       1 leaderelection.go:243] attempting to acquire leader lease  tigera-operator/operator-lock...
I0708 14:37:06.950468       1 leaderelection.go:253] successfully acquired lease tigera-operator/operator-lock
{"level":"info","ts":1625755026.9514308,"logger":"controller-runtime.manager.controller.tigera-installation-controller","msg":"Starting EventSource","source":"kind source: /, Kind="}
{"level":"info","ts":1625755027.0525558,"logger":"controller-runtime.manager.controller.tigera-installation-controller","msg":"Starting EventSource","source":"kind source: /V1, Kind=Secret"}
{"level":"info","ts":1625755027.154822,"logger":"controller-runtime.manager.controller.tigera-installation-controller","msg":"Starting EventSource","source":"kind source: /V1, Kind=ConfigMap"}
{"level":"info","ts":1625755027.256324,"logger":"controller-runtime.manager.controller.tigera-installation-controller","msg":"Starting EventSource","source":"kind source: /V1, Kind=ConfigMap"}
{"level":"info","ts":1625755027.2566109,"logger":"controller-runtime.manager.controller.tigera-installation-controller","msg":"Starting EventSource","source":"kind source: /, Kind="}
{"level":"info","ts":1625755027.3575156,"logger":"controller-runtime.manager.controller.tigera-installation-controller","msg":"Starting EventSource","source":"kind source: /, Kind="}
{"level":"info","ts":1625755027.4588928,"logger":"controller-runtime.manager.controller.tigera-installation-controller","msg":"Starting EventSource","source":"kind source: /, Kind="}
{"level":"info","ts":1625755027.559794,"logger":"controller-runtime.manager.controller.tigera-installation-controller","msg":"Starting EventSource","source":"kind source: /, Kind="}
{"level":"info","ts":1625755027.6607502,"logger":"controller-runtime.manager.controller.tigera-installation-controller","msg":"Starting EventSource","source":"kind source: /, Kind="}
{"level":"info","ts":1625755027.7658336,"logger":"controller-runtime.manager.controller.tigera-installation-controller","msg":"Starting EventSource","source":"kind source: /, Kind="}
{"level":"error","ts":1625755027.868342,"logger":"controller-runtime.source","msg":"if kind is a CRD, it should be installed before calling Start","kind":"APIService.apiregistration.k8s.io","error":"no matches for kind \"APIService\" in version \"apiregistration.k8s.io/v1beta1\"","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/go/pkg/mod/github.com/go-logr/zapr@v0.2.0/zapr.go:132\nsigs.k8s.io/controller-runtime/pkg/source.(*Kind).Start\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.7.0/pkg/source/source.go:117\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.7.0/pkg/internal/controller/controller.go:159\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.7.0/pkg/internal/controller/controller.go:205\nsigs.k8s.io/controller-runtime/pkg/manager.(*controllerManager).startRunnable.func1\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.7.0/pkg/manager/internal.go:691"}
{"level":"error","ts":1625755027.8686714,"logger":"setup","msg":"problem running manager","error":"no matches for kind \"APIService\" in version \"apiregistration.k8s.io/v1beta1\"","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/go/pkg/mod/github.com/go-logr/zapr@v0.2.0/zapr.go:132\nmain.main\n\t/go/src/github.com/tigera/operator/main.go:228\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:204"}```

**Comment by jayunit100** [comment]
I see the root cause... 
```
APIService
The apiregistration.k8s.io/v1beta1 API version of APIService will no longer be served in v1.22.

Migrate manifests and API clients to use the apiregistration.k8s.io/v1 API version, available since v1.10.
All existing persisted objects are accessible via the new API
No notable changes
```

**Comment by jayunit100** [comment]
sounds like this is fixed on master though

**Comment by tmjd** [comment]
Using the v1 of apiregistration has been in the operator since v1.18. This is also available in a Calico release v3.20 with operator v1.20 which was released a few days ago.
